### PR TITLE
Add COPA and SWAG

### PR DESCRIPTION
**Pull request by W4ngatang**
Add implementation of COPA and SWAG tasks

### CODE REVIEW

File: src/evaluate.py [hunk 1]
```diff
@@ -140,18 +141,21 @@ def write_preds(tasks: Iterable[tasks_module.Task], all_preds, pred_dir, split_n
...
             _write_commitment_preds(task, preds_df, pred_dir, split_name,
                                     strict_glue_format=strict_glue_format)
+        elif isinstance(task, COPATask):
+            _write_copa_preds(task, preds_df, pred_dir, split_name,
+                              strict_glue_format=strict_glue_format)
         else:
             log.warning("Task '%s' not supported by write_preds().",
                         task.name)
             continue
+        log.info("Task '%s': Wrote predictions to %s", task.name, pred_dir)
```
Comment (sleepinyourhat) [inline src/evaluate.py]: It looks like this'll appear twice now for Edge Probing, no?

File: src/models.py [hunk 2]
```diff
@@ -1110,6 +1122,40 @@ def _lm_forward(self, batch, task, predict):
             pass
         return out
 
+    def _mc_forward(self, batch, task, predict):
+        ''' Forward for a multiple choice question answering task '''
```
Comment (sleepinyourhat) [inline src/models.py]: How does this handle the context vs. question distinction? Are they just concatenated?

File: src/tasks/tasks.py [hunk 3]
```diff
@@ -2224,3 +2225,152 @@ def get_metrics(self, reset=False):
...
+
+        def _load_split(data_file):
+            contexts, questions, choicess, targs = [], [], [], []
+            data = [json.loads(l) for l in open(data_file, encoding="utf-8")]
+            for example in data:
+                context = example["premise"]
+                choice1 = example["choice1"]
+                choice2 = example["choice2"]
+                question = example["question"]
+                question = "What was the cause of this ?" if question == "cause" else "What happened as a result ?"
```
Comment (sleepinyourhat) [inline src/tasks/tasks.py]: Is the space before '?' intentional? (i.e., is this valid/expected with all the tokenizers we might use?)

File: src/tasks/tasks.py [hunk 4]
```diff
@@ -2224,3 +2225,152 @@ def get_metrics(self, reset=False):
...
+        instances = map(_make_instance, *split)
+        return instances
+
+    def get_metrics(self, reset=False):
+        '''Get metrics specific to the task'''
+        acc = self.scorer1.get_metric(reset)
+        return {'accuracy': acc}
+
+
+@register_task('swag', rel_path='SWAG/')
```
Comment (sleepinyourhat) [inline src/tasks/tasks.py]: Delete for now.
(Or just combine these PRs.)

File: src/models.py (hunk 2 above)
Comment (W4ngatang) [reply src/models.py]: Yea they're concatenated during preprocessing. The question and the answer choice get the same segment ID, which seems standard for BERT.

File: src/evaluate.py (hunk 1 above)
Comment (W4ngatang) [reply src/evaluate.py]: fixed

**Pull request merged**